service/controller/resource/tenantclients/create.go

```diff
@@ -52,13 +51,17 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		}
 
 		k8sClient, err = k8sclient.NewClients(c)
-		if tenant.IsAPINotAvailable(err) {
+		if err != nil {
+			// On any error we want to handle the situation gracefully in order
+			// to not block the whole reconciliation. Our former approach of
+			// matching against specific errors was extremely brittle because
+			// every now and then new errors popped up which we never knew or
+			// handled before. This is extremely painful after fact in a
+			// immutable infrastructure because it is super hard to fix once it
+			// breaks after it is released.
 			r.logger.LogCtx(ctx, "level", "debug", "message", "tenant API not available yet")
```

Comment:
```suggestion
			r.logger.LogCtx(ctx, "level", "debug", "message", "tenant API not available yet", "stack", microerror.JSON(err))
```

Maybe log the error stack so that when debugging why the TC API never gets there, one has more information to look further?

Reply:
Yes we should at least log the error here.